### ISSUE TITLE
[Problem] UserEditModes not passed to App::GeometryPython objects

### ISSUE THREAD

**Issue by Roy-043** [FreeCAD/FreeCAD#12124]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

UserEditModes are not passed to App::GeometryPython objects.

Steps to reproduce:
1. Edit Mod/Arch/ArchBuildingPart.py: Add `print(mode)` as the 1st line in the `setEdit` function.
2. Restart.
3. Switch to the Arch WB.
4. Create an [Arch_BuildingPart](https://wiki.freecad.org/Arch_BuildingPart).
5. Change [UserEditMode](https://wiki.freecad.org/Std_UserEditMode) to Cutting.
6. Double-click the BuildingPart in the Tree view.
7. Result: mode should be 2, but is zero.

### Full version info

```
OS: Windows 8 build 9600
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.35720 (Git)
Build type: Release
Branch: main
Hash: 00cd50bbf49c810555899efe112ec7595c76cd3a
Python 3.10.13, Qt 5.15.8, Coin 4.0.2, Vtk 9.2.6, OCC 7.6.3
Locale: Dutch/Netherlands (nl_NL)
Installed mods:
```

### Subproject(s) affected?

Core

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
@yorikvanhavre @Roy-043 is this still relevant for 1.0?

**Comment by luzpaz** [comment]
@yorikvanhavre is there anyone you suggest delegating this to ?

**Comment by yorikvanhavre** [comment]
Not really, it's definitely a core problem, but I'm not exactly sure where the underlying issue lies... Maybe @wwmayer would have an idea?

**Comment by wwmayer** [comment]
> Change [UserEditMode](https://wiki.freecad.org/Std_UserEditMode) to Cutting.

There was never any guarantee that a view provider must implement all the hard-coded edit modes. Especially Cutting doesn't make sense for part objects.

**Comment by Roy-043** [comment]
The specific edit mode in the example is just that: an example to demonstrate a bug.

**Comment by luzpaz** [comment]
GIven that, where are we at here ?

**Comment by Roy-043** [comment]
This is still an issue.

**Comment by adrianinsaval** [comment]
> There was never any guarantee that a view provider must implement all the hard-coded edit modes. Especially Cutting doesn't make sense for part objects.

What even is Cutting? And why do we have a harcoded list of edit modes that may not be available?

**Comment by Roy-043** [comment]
Still an issue.
```
OS: Ubuntu 22.04.5 LTS (ubuntu:GNOME/ubuntu/wayland)
Architecture: x86_64
Version: 1.1.0dev.42780 (Git)
Build date: 2025/07/28 17:17:34
Build type: Release
Branch: main
Hash: 8dc6397fc4433785f73f8a1958ff11243c348376
Python 3.11.13, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.2.5, IfcOpenShell 0.8.2, OCC 7.8.1
Locale: Dutch/Netherlands (nl_NL)
Navigation Style/Orbit Style/Rotation Mode: CAD/Free Turntable/Drag at cursor
Stylesheet/Theme/QtStyle: unset/FreeCAD Classic/fusion
Logical DPI/Physical DPI/Pixel Ratio: 96/86.1283/1
Installed mods:
```
